client/src/routes.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { Switch, Route } from "react-router-dom";
+import HomePage from "./containers/HomePage";
+import AddressPage from "./containers/AddressPage";
+import BBLPage from "./containers/BBLPage";
+import AboutPage from "./containers/AboutPage";
+import HowToUsePage from "./containers/HowToUsePage";
+import MethodologyPage from "./containers/Methodology";
+import TermsOfUsePage from "./containers/TermsOfUsePage";
+import PrivacyPolicyPage from "./containers/PrivacyPolicyPage";
+
+export const createRouteForAddressPage = (boro: string, streetname: string, housenumber?: string) =>
+  `/address/${boro}/${housenumber || " "}/${streetname}`;
```

Comment:
this is where you can just wrap the arguments in `encodeURIComponent()` to avoid having to wrap the whole URL in `encodeURI()`. (Honestly in practice I never use `encodeURI()`, only `encodeURIComponent()`.)